org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -130,7 +132,15 @@ public List<Either<Command, CodeAction>> getSourceActionCommands(CodeActionParam
 		boolean isInImportDeclaration =  CodeActionUtility.findASTNode(coveredNodes, coveringNode, ImportDeclaration.class) != null;
 
 		// Generate Constructor QuickAssist
-		if (isInFieldDeclaration || isInTypeDeclaration) {
+		boolean selectedStatic = false;
+		try {
+			selectedStatic = isSelectionStatic(context);
+		} catch (JavaModelException e1) {
+			// do nothing
+		}
+
+		// Generate Constructor QuickAssist
+		if (!selectedStatic && (isInFieldDeclaration || isInTypeDeclaration)) {

```

Comment:
I would move the static logic inside `getGenerateConstructorsAction`. You actually don't need to check for nodes on the position at which code actions are activated. You can just use the type passed into `getGenerateConstructorsAction` to get the fields, and for each field, I think you should be able to check its flag option, to determine if it's static.